gateway/manager/manager.go

```diff
@@ -190,16 +190,19 @@ func (s *Service) createHandler(schema *graphql.Schema) *graphqlHandler {
 
 func (s *Service) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
-	if (*r).Method == "OPTIONS" && s.appCfg.Cors.Enabled {
+	if s.appCfg.Cors.Enabled {
 		allowedOrigins := strings.Join(s.appCfg.Cors.AllowedOrigins, ",")
 		allowedHeaders := strings.Join(s.appCfg.Cors.AllowedHeaders, ",")
 		w.Header().Set("Access-Control-Allow-Origin", allowedOrigins)
 		w.Header().Set("Access-Control-Allow-Headers", allowedHeaders)
 		// setting cors allowed methods is not needed for this service,
 		// as all graphql methods are part of the cors safelisted methods
 		// https://fetch.spec.whatwg.org/#cors-safelisted-method
-		w.WriteHeader(http.StatusOK)
-		return
+
+		if (*r).Method == "OPTIONS" {
+			w.WriteHeader(http.StatusOK)
+			return
+		}
```

Comment:
It's not directly part of your changes, but I suggest to get rid of the unnecessary dereference and use the const for the method like so: 
```suggestion
		if r.Method == http.MethodOptions {
			w.WriteHeader(http.StatusOK)
			return
		}
```